### ISSUE TITLE
unable to find server because of ping filter

### ISSUE THREAD

**Issue by aman7** [SmartlyDressedGames/Unturned-3.x-Community#4650]
![ping](https://github.com/user-attachments/assets/1def2b88-0452-4dd6-ab13-d6e12162cf15)
1. players are unable to find US located server if they are playing from EU or SEA region. i know they can adjust the ping filter and find, but it will be difficult for new players (who just downloaded the game) to get familiar with these filters. 
2. Also it is pain for players who are returning back after 1-2 years and want to find their favorite server with name, they think servers have been shutdown.
3. Default ping is set to 200, which applies to All History, All Favorites, All friends, and just because of ping filter players are unable to locate the servers:
       a. they played before
       b. they favorites it
       c. if their friends are playing on it.

for example, if i played on a server before / marked it favorite / if my friends are playing on it. i cannot see those server if my ping is more than 200.
4. i know ping matters a lot in game like unturned, but for pve servers its fine to have high ping, many players on our server plays with 250+ ping without any issue.

**Comment by Jdance-Media** [comment]
I personally believe that the max ping filter should be disabled by default.

**Comment by SDGNelson** [comment]
There has been some contention about it from a few hosts.

1. If they just downloaded the game then I don't think a server in a different region is ideal. :S
2/3. Good point! We can change the default for the non-Internet presets to ping filter off.
4. For PvE it's true you can have an enjoyable time with a higher ping.

One reason I'm thinking about is if the server list can now default to sorting by player count rather than by ping since the servers at the top won't have *too high* of a ping. I wonder if this would help lead new players into populated servers.

@Jdance-Media Why do you prefer for it to be off by default?

**Comment by aman7** [comment]
1. i think region does matter for new players, people from India, indonesia, malaysiya, philipins, will see servers from china/russia, which they cant even read. 
2. right now i see only 10 pve server with default ping from my region.
3. since you are going to turn ping filter off for non-internet preset, i dont think it will help much with new players because on my servers players plays from all over the world. and i agree with Jdance-Media 
4. you can try default sorting by pop, but i am looking at my Russia map now, it has 4 players with high ping, the one with lowest ping has 0 players, and another one has 2 players.

**Comment by Jdance-Media** [comment]
> @Jdance-Media Why do you prefer for it to be off by default?

Simple, the filters you start out with by default should be the most broad. Even if the servers have an out-of-this-world 1000 ping, I still want to see them. A ping issue is somewhat easily recognizable, so if someone is having troubles with find close servers they'll turn the filter on themselves. I'd also like to add that the new filter did cause some confusion for me as well, as I wasn't seeing as many servers from across the pond.

**Comment by PandahutMushy** [comment]
Default ping absolutely needs to change to a higher value, at the very minimum. The points raised above by other users are all very valid, however, also consider that some networks cater to global audiences. Large networks are FULL of servers where North American and EU players battle it out. These are some of our most popular servers, both in our network and in Unturned as a whole.

My servers seen a reduction of 18-20% in player counts during the period that this change has been in effect. I don't have 100% certainly if the two things are related, but this has been a very frequent and constant complaint we've received from players who are usually new or getting started. 

I strongly recommend upping the default maximum ping. Players can always sort the list by lowest ping anyway. Plus, the large majority or new players (as evidenced by the frequent complaints about this filter), don't look or change filters until much later on, when they gain familiarity with the game.

**Comment by SDGNelson** [comment]
Some hopefully 🤞  good adjustments to this coming to the preview branch this evening and releasing tomorrow or Thursday:

- Anyone with the current default of 200 ms max ping will have it automatically raised to 300 ms, so more international servers are visible by default.
- The History, LAN, Favorites, and Friends presets have their defaults changed to 0 (off) so it doesn't apply there.
- Default server list sort no longer prioritizes ping as much. It will now weigh a mix of ping, fullness*, and player count.

*When considering fullness for this default sort, the max players is clamped to 100 since many servers with ~200 max players have < 100 players.

Here's how that stacks up at the moment:

![Screenshot of Unturned server browser](https://github.com/user-attachments/assets/bd286e22-bbde-46e5-974a-627d15ae0115)

**Comment by aman7** [comment]
1. i think 400 as default should be better.
i know you are concerned about players, who can find servers closer to them, i think i have some ideas.
i don't know if you have played Rust. i am posting some screenshots which might explain it better
![rust1](https://github.com/user-attachments/assets/d9965043-4912-44df-9c2e-a7e7c3eca428)

as i remember earlier you were concerned about servers with  donor stuff too, so maybe if servers meets your requirement and agrees to follow your guideline  you can show them under official (or keep it as Internet) totally depends on you, and rest as they are now.

![rust2](https://github.com/user-attachments/assets/457e6511-97d9-4a9c-8936-8fd5e3dffad8)

and for the ping issue they can choose a region if they want (from pic above), by default it will be off.
i have seen the same mechanism being used in 7 days to die.
i think with the region filter, we wont need to use ping filter.

**Comment by SDGNelson** [comment]
@aman7 I wonder if there's a hidden region filtering operation for Steam's server list API. It doesn't appear to be mentioned in: https://partner.steamgames.com/doc/api/ISteamMatchmakingServers

**Comment by aman7** [comment]
i am not sure about rust because when i used to run rust server there wasnt anything like that in config.
but in 7 days to die config file  we have something like 
<property name="Region" value="NorthAmericaWest" /> 

so i guess it is coming from here.

**Comment by PandahutMushy** [comment]
Nelson, could you please clarify what you mean here?

"*When considering fullness for this default sort, the max players is clamped to 100 since many servers with ~200 max players have < 100 players."

I don't see any servers capped at 100 in the screenshot you provided so I'm somewhat confused.

Thanks for considering our suggestions and implementing these changes.

**Comment by SDGNelson** [comment]
> i am not sure about rust because when i used to run rust server there wasnt anything like that in config. but in 7 days to die config file we have something like `<property name="Region" value="NorthAmericaWest" /> <!-- The region this server is in. Values: NorthAmericaEast, NorthAmericaWest, CentralAmerica, SouthAmerica, Europe, Russia, Asia, MiddleEast, Africa, Oceania -->`
> 
> so i guess it is coming from here.

Maybe it's implemented on a per-game basis then. I'd wondered if Steam was automatically categorizing them using a Geo-IP database. Maybe we should have an option for servers to specify their primary language and region. 🤔 

> @SDGNelson could you please clarify what you mean here?
> 
> "*When considering fullness for this default sort, the max players is clamped to 100 since many servers with ~200 max players have < 100 players."
> 
> I don't see any servers capped at 100 in the screenshot you provided so I'm somewhat confused.
> 
> Thanks for considering our suggestions and implementing these changes.

Sure thing: Although there are many servers with 200 max players set, typically they have 90-100 players at a time. I'd guess mostly just the game not supporting 100+ players very well. So even though these servers have tons of players, when measuring by fullness it'd be 40-50% full. The new default sort prioritizes servers around 75% full - the thought being that near-empty servers aren't a priority, and almost-full servers are probably interesting but might not have room for more players. (That said, servers around 50% and 100% full are only sorted slightly lower than 75% full.) Since the servers with 200 max players are still very full I clamped the fullness sorting max players to 100 so a server with 90 players and 200 max players is considered 90% full for sorting purposes.

**Comment by SDGNelson** [comment]
I wanted to check back these changes hadn't negatively affected players counts, and using the same tool as [my post earlier this year](https://blog.smartlydressedgames.com/2024/01/15/2023-server-list-stats/) it appears the player counts on public servers are about the same:
![playercount](https://github.com/user-attachments/assets/a9cd6b4a-1a03-49ca-a1e8-7cf3d19f395e)

**Comment by PandahutMushy** [comment]
Hi Nelson,

Just to clarify about our player counts going lower, this was very short-lived, as players adjusted to the filter changes. Your changes from August 21st surely helped as well. Our player base is roughtly 50% North American. On August 15th, we noticed that something was up due to several similar player reports and lower EU player counts.

![PlrCounts_NA-EU_Aug15](https://github.com/user-attachments/assets/99e5050d-3e7d-4c59-b378-c555fceb9bcc)

In the above graph, note that on August 15th the EU player numbers significantly dropped, while North American player counts increased (likely due to being more often suggested to the lion's share of the online player base). I imagine that American players represent roughly 50% of Unturned's total online players as well. This is evidenced by our CDN request numbers coming from the Unturned server browser, making requests to our server's favico image. 

![CDNRequests_Aug2024](https://github.com/user-attachments/assets/7e60fd90-9368-414b-8d9d-8a07e86af0c1)

Overall, Unturned's online population has been healthy and stable, for years now. The changes from August 15th would likely not have led to an inconvenience as large as some of us had imagined. However, last week's changes corrected most of our concerns. On my end, this issue has been resolved. Thank you for the awesome server browser upgrade.
